File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let levels = []
    function recurse(node, level) {
        if (levels.length === level) levels.push([])
        levels[level].push(node.val)
        if (node.left) recurse(node.left, level + 1)
        if (node.right) recurse(node.right, level + 1)
    }
    if (root) recurse(root, 0)
    return levels
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        res = []
        levels = []
        queue = deque([root])

        while queue:
            level = []
            n = len(queue)

            for _ in range(n):
                node = queue.popleft()

                level.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            levels.append(level)

        return levels